src/nlsolve/muller.jl

```diff
@@ -0,0 +1,48 @@
+"""
+    Muller()
+
+Muller's method.
+"""
+struct Muller <: AbstractSimpleNonlinearSolveAlgorithm end
+
+function SciMLBase.solve(prob::NonlinearProblem, alg::Muller, args...;
+    abstol = nothing, maxiters = 1000, kwargs...)
+    @assert !isinplace(prob) "`Muller` only supports OOP problems."
+    xᵢ₋₂, xᵢ₋₁, xᵢ = prob.u0
+    @assert xᵢ₋₂ ≠ xᵢ₋₁ ≠ xᵢ ≠ xᵢ₋₂
```

Comment:
```suggestion
    @assert xᵢ₋₂ ≠ xᵢ₋₁ ≠ xᵢ
```

Maybe check with an approx?

Reply:
The final inequality is needed to ensure that
```julia
xᵢ ≠ xᵢ₋₂
```
Muller's method requires all three initial values to be different.